client/core/simnet_trade.go

```diff
@@ -877,154 +849,173 @@ func testOrderStatusReconciliation(s *simulationTest) error {
 		return fmt.Errorf("client 2 locked funds not returned: locked before trading %v, locked after trading %v, "+
 			"locked after reconnect %v", preTradeLockedBalance, preDisconnectLockedBalance, c2Balance.Locked)
 	}
+	s.client2.enableWallets()
+
+	for _, c := range s.clients {
+		if err := c.mineMedian(context.TODO(), s.quote.id); err != nil {
+			return err
+		}
+		if err := c.mineMedian(context.TODO(), s.base.id); err != nil {
+			return err
+		}
+	}
+
+	s.waitALittleBit()
+
 	return nil
 }
 
-// TestResendPendingRequests runs a simple trade test, simulates init/redeem
-// request errors during trade negotiation and ensures that failed requests
-// are retried and the trades complete successfully.
-func testResendPendingRequests(s *simulationTest) error {
-	var qty, rate uint64 = 1 * s.lotSize, 250 * s.rateStep
-	s.client1.isSeller, s.client2.isSeller = true, false
+var _ comms.WsConn = (*tConn)(nil)
 
-	c1OrderID, c2OrderID, err := s.placeTestOrders(qty, rate)
+type tConn struct {
+	routeCh      chan string
+	errCh        chan error
+	on           uint32
+	closeAndWait func()
+
+	realConn comms.WsConn
+}
```

Comment:
I was toying with a different way.

https://github.com/buck54321/dcrdex/compare/a7ddea84e2ac6593270a496582474a765f675fde...buck54321:dcrdex:jg-simnet-test-simplification

I think I got the `bits` part wrong? I haven't quite wrapped my head around the `bits` yet, but the approach seems simpler.

Reply:
That's not bad.  Not sure about bits either.
Would there be a data race on these assignments, or are comms at this points?

```go
	s.client1.filteredConn.requestFilter = preFilter
	s.client2.filteredConn.requestFilter = preFilter

	defer func() {
		s.client1.filteredConn.requestFilter = nil
		s.client2.filteredConn.requestFilter = nil
	}()
	}()
```

Reply:
I think you're right. I pushed an update.

Reply:
Seems like feasibility of this alt approach depends on intent of bits.  Commented here: https://github.com/decred/dcrdex/pull/1909#discussion_r1004991954